### ISSUE TITLE
Bug: if you set the value of an observable inside a subscribe that depends on that observable, the changes won't get propagated to pure computeds

### ISSUE THREAD

**Issue by meltuhamy** [knockout/knockout#2282]
```js
const value = ko.observable('hello').extend({notify: 'always'});
const value_readOnly = ko.pureComputed(() => value()).extend({notify: 'always'});

value.subscribe(v => document.write(`<p>value: ${v}</p>`));
value_readOnly.subscribe(v => document.write(`<p>value_readOnly: ${v}</p>`));

// someone might change value in a subscribe!
value_readOnly.subscribe(v => {
  if(v%5===0 ){
    value('🐛')
  }
});


value(1);
value(2);
value(5); //bug!
```

Reproduced here: https://plnkr.co/edit/UtsjOzUaJkuQRaU338wF?p=preview

**Comment by chrisknoll** [comment]
Don't they say pure computeds should not have side effects? Is setting another observable in a computed considered a 'side effect'?

**Comment by meltuhamy** [comment]
I'm not setting the variable inside a computed

**Comment by chrisknoll** [comment]
Still a side-effect, I think, even if it's done in a subscribe of the computed.  I changed your code to this:

const value_readOnly = ko.computed(() => value()).extend({notify: 'always'});


and the output shows the correct value.  Maybe this is a bug with pureComputed, maybe this is a known limitation....

**Comment by mbest** [comment]
> I changed your code to this: ... and the output shows the correct value

I doesn't work with `ko.computed` either as far as I can tell. But updating the Knockout version to 3.4.2 and using `ko.options.deferUpdates = true;` does make it work.

**Comment by chrisknoll** [comment]
that's odd, i really did change that code above in his plunker and re-ran it and saw the correct output. Strange!

**Comment by meltuhamy** [comment]
@mbest thanks for that. I'm interested to know if there are any implications of setting deferUpdates to true... I'm guessing it means updates to observables are no longer synchronous?

**Comment by fastfasterfastest** [comment]
> I'm interested to know if there are any implications of setting deferUpdates to true... 

Yes, there are implications - serious testing of your application is suggested.  There are also updates/fixes waiting in the pipeline for next version (3.5.0) that affects the behavior and performance when `deferUpdates` is `true`.

**Comment by mbest** [comment]
The issue described here goes way back to version 2.1 where we decided to prevent `ko.computed` from being recursively updated. This was done because the dependency detection code had re-entrancy problems. Later, those problems were fixed, but allowing recursion would be a breaking change, one which we decided not to make.

**Comment by mbest** [comment]
Related: #1905
